front/src/applications/operationalStudies/hooks/useSetupItineraryForTrainUpdate.ts

```diff
@@ -92,7 +92,8 @@ const useSetupItineraryForTrainUpdate = (
                   (suggestedOp) =>
                     'uic' in step &&
                     suggestedOp.uic === step.uic &&
-                    suggestedOp.ch === step.secondary_code
+                    // If 'secondary_code' is not in step, return true; if 'secondary_code' is in step, check if suggestedOp.ch matches step.secondary_code
+                    (!step.secondary_code || suggestedOp.ch === step.secondary_code)
```

Comment:
Rather than explaining the code, I would suggest this, is that ok for you ?

```suggestion
                    // When importing train from open data or from files, secondary_code might not always exist
                    (!step.secondary_code || suggestedOp.ch === step.secondary_code)
```

Reply:
Sounds good to me, Fixed.